components/block/element.js

```diff
@@ -9,61 +10,119 @@ export default class Block extends LitElement {
   static styles = [style];
 
   static properties = {
-    blockId: { type: Number},
+    loading: {type: Boolean, state: true},
+    blockId: {type: String},
   };
 
-  renderMenu(){
-    return html`
-      <nav>
-        <button @click=${this.edit}>Edit</button>
-        <button @click=${this.moveUp}>Move Up</button>
-        <button @click=${this.moveDown}>Move Down</button>
-        <button @click=${this.delete}>Delete</button>
-      </nav>
-    `;
+  constructor() {
+    super();
+
+    this.loading = false;
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    this.model.on('sidebar_save:success', this.refresh.bind(this));
+  }
+
+  get layout() {
+    return window.parent.Core.g.layout;
+  }
+
+  get model() {
+    return (this.cached_model ||= this.layout.blocks.findWhere({
+      id: this.blockId,
+    }));
+  }
+
+  get parentModel() {
+    if (this.cached_parent_model !== undefined) return this.cached_parent_model;
+
+    const parentId = this.model.attributes.parent_block_id;
+    const parentModel = document.querySelector(
+      `ngl-block[blockId="${parentId}"]`
+    );
+
+    if (parentModel && isBlock(parentModel))
```

Comment:
Why do we need to test isBlock here when we are always searching for `ngl-block` which is instance of `Block`?

Reply:
You are right, I don't think it is necessary. Removed the test in fe0b4ee3e33c6f4ff7dfae10e69e64f68266fe59

Reply:
Great, then you can also remove `isBlock` function on top

Reply:
Yep, I missed that. Removed `isBlock` in f4602e9bd9225f12b6fb88d62b34ba95b82b62eb